modules/apps/transfer/keeper/relay.go

```diff
@@ -277,6 +277,10 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		return errorsmod.Wrap(err, "failed to mint IBC tokens")
 	}
 
+	if !k.bankKeeper.HasDenomMetaData(ctx, voucherDenom) {
```

Comment:
Not strictly related to my proposal of setting denom trace and metadata simultaneously, I realize this code placement is quite odd. It is in between two actions that must be atomic (mint and send minted tokens to receiver).  I would prefer it to go right after the `HasDenomTrace` function

Reply:
I moved the code up. If it's ok with you, maybe we can address in a different issue/PR performing `SetDenomTrace` and `setDenomMetadata` actions in the same function call.

Reply:
sure I don't feel strongly about it